### ISSUE TITLE
您好，我使用您的数据集进行语义分割任务，尝试了论文中提到的PSPNEt，OCRnet kNET等网络可是对于前景（缺陷类别）都不能进行正确分类，是需要进行什么处理吗

### ISSUE THREAD

**Issue by Donar7** [wch313/PSCDE-Dataset#3]
![捕获](https://github.com/wch313/PSCDE-Dataset/assets/67570657/b1e69b0c-daee-4ac6-9535-6d2369885c98)

**Comment by wch313** [comment]
推荐使用Dice loss和adam优化器再次尝试；同时还有把忽略固定标签值的设置关闭掉

**Comment by Donar7** [comment]
![捕获2](https://github.com/wch313/PSCDE-Dataset/assets/67570657/81a8ca4e-e7a7-4c6d-8ef5-958e7bc48517)
您好，不好意思再次打扰您，我根据您的建议尝试了多次之后还是不能正确
![微信图片_20240516093529](https://github.com/wch313/PSCDE-Dataset/assets/67570657/1e2f78af-db8c-4273-bc84-94472f860ac6)
完成缺陷的训练，这是我的参数设置，同时我也关闭了忽略固定标签，结果是这样![微信图片_20240516093529](https://github.com/wch313/PSCDE-Dataset/assets/67570657/1e2f78af-db8c-4273-bc84-94472f860ac6)

**Comment by wch313** [comment]
GT缺陷像素值是否设置为1？ 需要自行转化一下
import os
import cv2
import numpy as np

def convert_to_heiBai(image_path,save_path):
    for img in os.listdir(image_path):

        if img == "desktop.ini":
            print("desktop.ini")
        else:
          image = cv2.imread(image_path+img,cv2.IMREAD_GRAYSCALE)
        # 在这里设置阈值
          a = np.where(image == 255, 1 ,0) #等于255的变为1 其余的为0
          a = np.array(a, dtype=np.uint8)
          cv2.imwrite(save_path+img,a)
          print(img)
if __name__ == '__main__':
    image_path =  ''
    save_path  =  ''

    if not os.path.exists(save_path):
        os.makedirs(save_path)
    convert_to_heiBai(image_path,save_path)

**Comment by Donar7** [comment]
检查过了，确认已经把mask的值设为1了，这是我的训练日志，不知道哪里出问题了
![捕获3](https://github.com/wch313/PSCDE-Dataset/assets/67570657/8d4f7748-733d-4c64-929f-2c4f0c130dc9)

**Comment by wch313** [comment]
reduce_zero_label这个参数取个反False

**Comment by Donar7** [comment]
![捕获4](https://github.com/wch313/PSCDE-Dataset/assets/67570657/b14210e0-2a36-4a6d-9d68-641c588180c5)
一直是FALSE阿，刚才Ture也试过了还是不行

**Comment by wch313** [comment]
norm_cfg = dict(type='SyncBN', requires_grad=True)
model = dict(
    type='CascadeEncoderDecoder',
    num_stages=2,
    pretrained='open-mmlab://msra/hrnetv2_w48',
    backbone=dict(
        type='HRNet',
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        norm_eval=False,
        extra=dict(
            stage1=dict(
                num_modules=1,
                num_branches=1,
                block='BOTTLENECK',
                num_blocks=(4, ),
                num_channels=(64, )),
            stage2=dict(
                num_modules=1,
                num_branches=2,
                block='BASIC',
                num_blocks=(4, 4),
                num_channels=(48, 96)),
            stage3=dict(
                num_modules=4,
                num_branches=3,
                block='BASIC',
                num_blocks=(4, 4, 4),
                num_channels=(48, 96, 192)),
            stage4=dict(
                num_modules=3,
                num_branches=4,
                block='BASIC',
                num_blocks=(4, 4, 4, 4),
                num_channels=(48, 96, 192, 384)))),
    decode_head=[
        dict(
            type='FCNHead',
            in_channels=[48, 96, 192, 384],
            channels=720,
            input_transform='resize_concat',
            in_index=(0, 1, 2, 3),
            kernel_size=1,
            num_convs=1,
            norm_cfg=dict(type='SyncBN', requires_grad=True),
            concat_input=False,
            dropout_ratio=-1,
            num_classes=2,
            align_corners=False,
            loss_decode=dict(
                type='DiceLoss', use_sigmoid=False, loss_weight=0.4)),
        dict(
            type='OCRHead',
            in_channels=[48, 96, 192, 384],
            channels=512,
            ocr_channels=256,
            input_transform='resize_concat',
            in_index=(0, 1, 2, 3),
            norm_cfg=dict(type='SyncBN', requires_grad=True),
            dropout_ratio=-1,
            num_classes=2,
            align_corners=False,
            loss_decode=dict(
                type='DiceLoss', use_sigmoid=False, loss_weight=1.0))
    ],
    train_cfg=dict(),
    test_cfg=dict(mode='whole'))
dataset_type = 'ADE20KDataset'
data_root = 'data/ade/ADEChallengeData2016'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
crop_size = (512, 512)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=False),
    dict(type='Resize', img_scale=(512, 512), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(512, 512),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=4,
    workers_per_gpu=4,
    train=dict(
        type='ADE20KDataset',
        data_root='data/ade/ADEChallengeData2016',
        img_dir='images/training',
        ann_dir='annotations/training',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', reduce_zero_label=False),
            dict(type='Resize', img_scale=(512, 512), ratio_range=(0.5, 2.0)),
            dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='ADE20KDataset',
        data_root='data/ade/ADEChallengeData2016',
        img_dir='images/validation',
        ann_dir='annotations/validation',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(512, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='ADE20KDataset',
        data_root='data/ade/ADEChallengeData2016',
        img_dir='images/validation',
        ann_dir='annotations/validation',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(512, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
log_config = dict(
    interval=50, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
cudnn_benchmark = True
optimizer = dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=0.0005)
optimizer_config = dict()
lr_config = dict(policy='poly', power=0.9, min_lr=1e-08, by_epoch=False)
runner = dict(type='IterBasedRunner', max_iters=160000)
checkpoint_config = dict(by_epoch=False, interval=2000)
evaluation = dict(interval=2000, metric='mFscore', pre_eval=True)
work_dir = './work_dirs/ocrnet_hr48_512x512_80k_ade20k'
gpu_ids = [0]
auto_resume = False


仅供参考，这是我训练用的参数

**Comment by wch313** [comment]
mmseg需要设置的东西挺多的，还有一些定义啥的，你再多尝试尝试。数据集设置，我直接使用了ADE20KDataset的设置，并对应需改为了PSCDE数据集

**Comment by wch313** [comment]
我的这个mm版本也比较旧

**Comment by wch313** [comment]
adam的学习率我会设置的再小一些10-4/10-5

**Comment by Donar7** [comment]
太感谢了！试了很多次总算有结果了！谢谢您！祝您一切顺利

**Comment by wch313** [comment]
不客气！祝您科研顺利，paper多多！！
